### PR TITLE
Print  the full qualified name of the variable type when the parent of the type has generics

### PR DESCRIPTION
**Pull request by tdurieux**

### CODE REVIEW

File: src/main/java/spoon/reflect/visitor/DefaultJavaPrettyPrinter.java [hunk 1]
```diff
@@ -1891,7 +1891,8 @@ public void visitCtCircularTypeReference(CtCircularTypeReference reference) {
 
 	private <T> boolean hasDeclaringTypeWithGenerics(CtTypeReference<T> reference) {
 		// If current reference use generic types, we don't need this hack.
-		if (reference.getActualTypeArguments().size() != 0) {
+		if (reference.getActualTypeArguments().size() != 0
+				&& reference.getDeclaringType().getActualTypeArguments().size() == 0) {
```
Comment (GerardPaligot) [inline src/main/java/spoon/reflect/visitor/DefaultJavaPrettyPrinter.java]: Can you move this check below this [check](https://github.com/INRIA/spoon/pull/389/files#diff-5f33bf5d1f38230de815e4c60a67a215R1906)?

Comment (tdurieux) [inline src/main/java/spoon/reflect/visitor/DefaultJavaPrettyPrinter.java]: The complete if or just my additional condition?

Comment (GerardPaligot) [inline src/main/java/spoon/reflect/visitor/DefaultJavaPrettyPrinter.java]: Your additional condition. Potentially, `reference.getDeclaringType()` can be null.

Comment (tdurieux) [inline src/main/java/spoon/reflect/visitor/DefaultJavaPrettyPrinter.java]: No, I cannot. but I can move the complete if

Comment (GerardPaligot) [inline src/main/java/spoon/reflect/visitor/DefaultJavaPrettyPrinter.java]: Move the complete if.

**Pull request merged**